presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorAnalyzeTableMetadata.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.connector;
+
+import com.facebook.presto.spi.ConnectorTableHandle;
+import com.facebook.presto.spi.ConnectorTableMetadata;
+import com.facebook.presto.spi.statistics.TableStatisticsMetadata;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorAnalyzeTableMetadata
+{
+    private final TableStatisticsMetadata statisticsMetadata;
+    private final ConnectorTableMetadata tableMetadata;
+    private final ConnectorTableHandle tableHandle;
+
+    public ConnectorAnalyzeTableMetadata(TableStatisticsMetadata statisticsMetadata, ConnectorTableMetadata tableMetadata, ConnectorTableHandle tableHandle)
+    {
+        this.statisticsMetadata = requireNonNull(statisticsMetadata, "statisticsMetadata is null");
+        this.tableMetadata = requireNonNull(tableMetadata, "tableMetadata is null");
+        this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+    }
+
+    public TableStatisticsMetadata getStatisticsMetadata()
+    {
+        return statisticsMetadata;
+    }
+
+    public ConnectorTableMetadata getTableMetadata()
```

Comment:
I used that mainly to save a `getTableMetadata()` call when planning table scan for the analyze statement. The connector will need to call `getTableMetadata()` when generating `TableStatisticsMetadata`. 

Alternatively, we can add `ColumnMetadata` to `TableStatisticsMetadata` to describe the property of columns that we are collecting statistics for. Then we can use these information when planning the table scan.

I don't see any use case that we want to return different `TableMetadata` for analyze statement.

Reply:
The idea is that, when generating the `TableStatisticsMetadata`, the connector has already done a `getTableMetadata()` which contains all the expensive operations. And in order to plan the table scan for `ANALYZE`, we need a list of column handle which is not in `TableStatisticsMetadata`. So I included a `TableMetadata` object in the `AnalyzeTableMetadata` to reuse it and avoid one more call. 

Like I said, an alternative is to include the column information in `TableStatisticsMetadata` to describe the properties of columns we are collecting statistics for. So that later we can use it to plan table scan for `ANALYZE`.  Any thoughts?

Reply:
I don't feel strong about it. You can keep metadata in place if you would like.